PWGEM/PhotonMeson/TableProducer/skimmerGammaConversions.cxx

```diff
@@ -277,18 +314,36 @@ struct skimmerGammaConversions {
 
     // none of tracks has a mother, has been accounted for in fMotherSizesHisto
     if ((lMothersIndecesPos.size() + lMothersIndecesNeg.size()) == 0) {
+      hasMCParticle = false;
+
       return kNoTrackComesFromMother;
     }
 
     // exactly one track has one mother
     if ((lMothersIndecesPos.size() + lMothersIndecesNeg.size()) == 1) {
+      hasMCParticle = false;
+
       return kOneTrackHasOneMother;
     }
 
     // we know now both tracks have at least one mother
     // check if it is the same
     if (lMothersIndecesPos[0] != lMothersIndecesNeg[0]) {
+      // fill Track Mc true table
+      hasMCParticle = true;
+      fillfFuncTableMCTrackInformation(theTrackPos, false);
+      lPosEntryInMCTrack = fFuncTableMCTrackInformation.lastIndex();
+      fillfFuncTableMCTrackInformation(theTrackNeg, false);
+      lNegEntryInMCTrack = fFuncTableMCTrackInformation.lastIndex();
+
       return kNotSameMothers;
+    } else {
+      // fill Track Mc true table
+      hasMCParticle = true;
+      fillfFuncTableMCTrackInformation(theTrackPos, true);
+      lPosEntryInMCTrack = fFuncTableMCTrackInformation.lastIndex();
+      fillfFuncTableMCTrackInformation(theTrackNeg, true);
```

Comment:
you need this whole block only once. And also, since I think you want to fill the v0daughterMcParticles table in any case when mc particles exist for the daughters, you need to do the following.
Instead of the return statements in lines 319, 326 and 339 you store these return reasons to a local variable. Then when you finally know, if the tracks have the same mother you can make the fillfFuncTableMCTrackInformation(, ) calls where the second argument will be a variable. And then you return with  the return reason from your local variable.